pyproject.toml

```diff
@@ -196,3 +197,44 @@ optional_poetry_groups = ["dev"]
 poetry_extras = ["file-based", "vector-db-based"]
 poe_tasks = ["check-ci"]
 mount_docker_socket = true
+
+[tool.deptry.per_rule_ignores]
+# This is a mapping of rules and package names to be ignored for that rule.
+DEP001 = [
+  # Optional dependencies that might be imported conditionally
+  "dagger",  # Used in bin scripts but not a core dependency
+  "source_declarative_manifest"  # Used in tests but not a core dependency
+]
+DEP004 = [
+  "pdoc",  # Only used for generating documentation. Not a runtime dependency.
+  "pytest",  # Test dependencies
+  "freezegun",  # Test dependencies
+  "requests_mock",  # Test dependencies
+  "_pytest",  # Test dependencies
+  "pympler"  # Test dependencies
+]
```

Comment:
Can you check if any main code (not tests or supporting scripts) depend on these imports?